crates/path_serde_derive/src/container.rs

```diff
@@ -0,0 +1,109 @@
+use proc_macro_error::abort;
+use syn::{
+    parenthesized, parse::Parse as _, punctuated::Punctuated, Data, DataStruct, DeriveInput, Ident,
+    Result, Token, Type, WherePredicate,
+};
+
+pub struct Container {
+    pub fields: Vec<Field>,
+    pub bounds: Option<Punctuated<WherePredicate, Token![,]>>,
+    pub computed_leafs: Vec<ComputedLeaf>,
+}
+
+impl Container {
+    pub fn try_from_ast(item: &DeriveInput) -> Result<Self> {
+        let mut bounds = None;
+        let mut computed_leafs = Vec::new();
+
+        for attribute in &item.attrs {
+            if !attribute.path().is_ident("path_serde") {
+                continue;
+            }
+            attribute.parse_nested_meta(|meta| {
+                if meta.path.is_ident("bound") {
+                    let value = meta.value()?;
+                    bounds = Some(value.parse_terminated(WherePredicate::parse, Token![,])?);
+                    Ok(())
+                } else if meta.path.is_ident("add_leaf") {
+                    let content;
+                    parenthesized!(content in meta.input);
+                    let identifier: Ident = content.parse()?;
+                    content.parse::<Token![:]>()?;
+                    let into_type = content.parse::<Type>()?;
+                    computed_leafs.push(ComputedLeaf {
+                        identifier,
+                        into_type,
+                    });
+                    Ok(())
+                } else {
+                    Err(meta.error("unknown attribute"))
+                }
+            })?;
+        }
+
+        let fields = match &item.data {
+            Data::Struct(data) => read_fields(data)?,
+            Data::Enum(..) => Vec::new(),
+            Data::Union(..) => Vec::new(),
+        };
+        Ok(Container {
+            computed_leafs,
+            bounds,
+            fields,
+        })
```

Comment:
Change this order to be in line with the struct definition.

Reply:
done